tests/test_modeling_utils.py

```diff
@@ -1823,6 +1823,16 @@ def test_error_no_flash_available(self):
 
         self.assertTrue("does not support Flash Attention 2.0" in str(cm.exception))
 
+    def test_error_no_flash_available_with_config(self):
```

Comment:
Can you add a test for e.g. llama + passing a config + `attn_implementation="flash_attention_2`?

Reply:
You mean without `AutoModel`?

Reply:
I mean a test for an architecture that do support FA2, passing both a config + `attn_implementation="flash_attention_2"`

Reply:
Done!